apis/python/src/tiledbsoma/experiment_query/query.py

```diff
@@ -0,0 +1,503 @@
+import concurrent.futures
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ContextManager,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+import anndata
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pyarrow as pa
+from typing_extensions import Literal, TypedDict
+
+from .. import somacore  # to be replaced by somacore package, when available
+from ..dataframe import DataFrame as SOMADataFrame
+from ..sparse_nd_array import SparseNDArrayRead
+
+if TYPE_CHECKING:
+    from ..experiment import Experiment
+from ..sparse_nd_array import SparseNDArray as SOMASparseNDArray
+from .anndata import _make_anndata
+from .axis import AxisQuery
+from .eq_types import AxisColumnNames, ExperimentAxisQueryReadArrowResult
+
+AxisJoinIds = TypedDict(
+    "AxisJoinIds",
+    {
+        "obs": pa.Array,
```

Comment:
Here you have `pa.Array` not `Union[None, pa.Array` -- but at line 97 there is

```
        self._joinids = {
            "obs": None,
            "var": None,
        }
```

I don't know why the linter isn't saying anything ... 🤔

Reply:
It doesn’t know about any pyarrow types, so it just makes them all `Any`s.

Reply:
OK

Even for readability I think it would make sense to type these as `Union[None, pa.Array]`

Reply:
I will add the optional typing to the declaration, for clarity.